nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,59 @@
+const express = require("express");
+const { request } = require("../app");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  const queryParams = request.query;
+  const maxPrice = Number(request.query.maxPrice);
+  const limit = Number(request.query.limit);
+  const title = request.query.title;
+  const createdAfter = request.query.createdAfter;
+  // Check if a query parameter exist
+  if (queryParams) {
+    // Get meals that has a price smaller than maxPrice
+    if (maxPrice) {
+      response.send(meals.filter((meal) => meal.price <= maxPrice));
+      // Get meals that partially match a title
+    } else if (title) {
+      const filtredMealsByTitle = meals.filter((meal) =>
+        meal.title.toLowerCase().includes(title.toLowerCase())
+      );
+      if (filtredMealsByTitle.length === 0) {
+        response.send("Meal wasn't found");
+      } else {
+        response.send(filtredMealsByTitle);
+      }
+      // Get meals that has been created after the date
+    } else if (createdAfter) {
+      if (new Date(createdAfter) != "Invalid Date") {
+        response.send(
+          meals.filter(
+            (meal) => new Date(meal.createdAt) > new Date(createdAfter)
+          )
+        );
+      } else {
+        response.status(400).send("Invalid Date");
+      }
+      // Only specific number of meals
+    } else if (limit) {
+      response.send(meals.slice(0, limit));
+    } else response.status(400).send("Invalid query parameter");
+  } else {
+    response.send(meals);
+  }
+});
```

Comment:
What would happen if there is more than one valid parameter in the request? For example both limit and maxPrice. do we take limit value into account?

See https://hackyourfuture-cph.slack.com/archives/C02JMMMB0R3/p1649605920084789